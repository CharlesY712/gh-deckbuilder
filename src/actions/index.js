export const addCard = (card) => ({
  type: 'ADD_CARD',
  card
});

export const addCards = (cards) => ({
  type: 'ADD_CARDS',
  cards
});

export const removeCards = (cards) => ({
  type: 'REMOVE_CARDS',
  cards
});

export const addSelectedCard = (selectedCard) => ({
  type: 'ADD_SELECTED_CARD',
  selectedCard
});

export const addSelectedCards = (selectedCards) => ({
  type: 'ADD_SELECTED_CARDS',
  selectedCards
});

export const removeSelectedCard = (selectedCard) => ({
  type: 'REMOVE_SELECTED_CARD',
  selectedCard
});

export const removeSelectedCards = (selectedCards) => ({
  type: 'REMOVE_SELECTED_CARDS',
  selectedCards
});

export const addAvailableCard = (availableCard) => ({
  type: 'ADD_AVAILABLE_CARD',
  availableCard
});

export const addAvailableCards = (availableCards) => ({
  type: 'ADD_AVAILABLE_CARDS',
  availableCards
});

export const removeAvailableCard = (availableCard) => ({
  type: 'REMOVE_AVAILABLE_CARD',
  availableCard
});

