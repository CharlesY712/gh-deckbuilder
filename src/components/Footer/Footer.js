import React from 'react';
import './Footer.css';

export const Footer = () => {
  return (
    <footer>
      <h1>Select your deck here.</h1>
    </footer>
  );
};