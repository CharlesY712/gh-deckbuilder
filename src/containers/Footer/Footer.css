footer {
  text-align: center;
  width: 100vw;
  max-height: 65px;
  background-color: transparent;
  position: absolute;
  bottom: 0px;
  color: white;
  font-family: "HighTower";
  transition: all 1s;
}

footer:hover {
  max-height: 50vh;
}

.saved-decks-tab {
  width: 190px;
  margin: auto;
  background: rgba(210, 210, 210, 0.3);
  border-radius: 5px;
  position: relative;
  top: 15px;
}

.saved-decks-title {
  font-size: 26px;
}

.saved-decks-container {
  width: 85vw;
  margin: auto;
  display: flex;
  flex-wrap: wrap;
  justify-content: baseline;
  background-color: black;
  overflow-y: scroll;
}

#decks-container {
  display: flex;
  flex-wrap: wrap;
  justify-content: space-around;
  align-items: left;
  width: 80vw;
}

#decks-container h1 {
  width: 20vw;
  white-space: nowrap;
}

#decks-container img {
  position: relative;
  top: 10px;
}

.saved-deck {
  flex-basis: 42.5%;
  white-space: pre;
}

.saved-deck-classImg {
  width: 50px;
  height: 50px;
  display: inline-block;
}

.saved-deck-name {
  font-size: 26px;
  display: inline-block;
  line-height: 50px;
  margin: 0;
  text-decoration: none;
}

#decks-container button {
  background-color: transparent;
  border: none;
}

.saved-deck-deleteImg {
  width: 50px;
  height: 50px;
  display: inline-block;
}

.sign-in-request {
  margin-bottom: 0;
  flex-basis: 100%;
}

.sign-out {
  cursor: pointer;
  flex-basis: 100%;
}