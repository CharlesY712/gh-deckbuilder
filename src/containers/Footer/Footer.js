import React, { Component } from 'react';
import { Link, withRouter } from 'react-router-dom';
import { connect } from 'react-redux';
import * as actions from '../../actions';
import * as api from '../../api/index';
import './Footer.css';

export class Footer extends Component {
  constructor(props) {
    super(props);
    this.state = {
      decks: [],
      error: ''
    };
  };

  async componentDidMount() {
    try {
      const decks = await api.fetchDecks();
      this.setState({ decks })
    } catch (error) {
      this.setState({ error })
    }
  }

  async componentDidUpdate(prevProps) {
    console.log(this.props);
    
    if (prevProps !== this.props) {
      try {
        const decks = await api.fetchDecks();
        this.setState({ decks })
      } catch (error) {
        this.setState({ error })
      }
    }
  }

  mapDecks = (decks) => {
    const deleteImg = require('../../images/red-x.png')
    const mappedDecks = decks.map(deck => {
      const dynamicIcon = require(`../../images/classIcons/${deck.class}Icon.png`)
      const dynamicPath = `/${deck.class}`
    
      return (
        <div className="saved-deck" key={deck.id}>
          <Link to={dynamicPath}
          onClick={() => this.props.addSelectedDeck(deck.id)}
          >
          <img  className="saved-deck-classImg" src={dynamicIcon} alt={deck.class}/>
          <h1 className="saved-deck-name">{deck.name}</h1>
          </Link>
          <button>
            <img className="saved-deck-deleteImg"
              src={deleteImg}
              alt="delete"
              onClick={() => this.deleteDeck(deck.id)}
              />
          </button>
        </div>
      )
    })
    return mappedDecks;
  }

  addSelectedId = (deckId) => {
    this.props.addSelectedDeckId(deckId)
  }

  deleteDeck = async (deckId) => {
    try {
      await api.fetchDeleteDeck(deckId)
      const decks = await api.fetchDecks();
      this.setState({ decks })
    } catch (error) {
      this.setState({error})
    }
  }

  render() {
    return (
      <footer>
        <div className="saved-decks-tab">
          <h1 className="saved-decks-title">SAVED DECKS</h1>
        </div>
        <div className="saved-decks-container">
          {this.mapDecks(this.state.decks)}
        </div>
      </footer>
    );
  }
};

export const mapDispatchToProps = dispatch => ({
  addSelectedDeck: deckId => dispatch(actions.addSelectedDeck(deckId))
});

export default withRouter(connect(null, mapDispatchToProps)(Footer));