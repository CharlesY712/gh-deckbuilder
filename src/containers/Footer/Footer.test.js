import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import { Footer } from './Footer';
import { mapStateToProps, mapDispatchToProps } from './Footer';
import * as actions from '../../actions';
import * as mocks from '../../mocks/mockCards';

describe("Footer component", () => {
  describe("Container Tests", () => {
    let wrapper;

    beforeEach(() => {
      wrapper = shallow(<Footer
        addDecks={jest.fn()}
        addSelectedDecks={jest.fn()}
        changeUser={jest.fn()}
        currentDecks={[]}
        user="guest"/>)
    });

    it("should match the snapshot without being signed in", () => {
      expect(wrapper).toMatchSnapshot();
    });

    it("should match the snapshot when a user is signed in", () => {
      wrapper.setState({isSignedIn: true});

      expect(wrapper).toMatchSnapshot();
    });

    it("should match snapshot when there are decks", () => {
      wrapper = shallow(<Footer
        addDecks={jest.fn()}
        addSelectedDecks={jest.fn()}
        changeUser={jest.fn()}
        currentDecks={mocks.mockMultiDecks}/>)

      expect(wrapper).toMatchSnapshot();
    });
  });

  describe("Map State to Props", () => {
      it('correctly maps currentDecks to props', () => {
        const mockState = {currentDecks: mocks.mockMultiDecks, user:"guest"};
        const mapped = mapStateToProps(mockState);
        const expectedState = mocks.mockMultiDecks;

        expect(mapped.currentDecks).toEqual(expectedState);
      });

      it('correctly maps user to props', () => {
        const mockState = {currentDecks: mocks.mockMultiDecks, user:"guest"};
        const mapped = mapStateToProps(mockState);
        const expectedState = "guest";

        expect(mapped.user).toEqual(expectedState);
      });
  });

  describe("Map Dispatch to Props", () => {
    let mockDispatch;
    let mappedDispatch;

    beforeEach(() => {
      mockDispatch = jest.fn();
      mappedDispatch = mapDispatchToProps(mockDispatch);
    });

    it('Should dispatch addSelectedDeck', () => {
      mappedDispatch.addSelectedDeck();
      const expected = actions.addSelectedDeck();
      expect(mockDispatch).toHaveBeenCalledWith(expected);
    });

    it('Should dispatch addDecks', () => {
      mappedDispatch.addDecks();
      const expected = actions.addDecks();
      expect(mockDispatch).toHaveBeenCalledWith(expected);
    });

    it('Should dispatch changeUser', () => {
      mappedDispatch.changeUser();
      const expected = actions.changeUser();
      expect(mockDispatch).toHaveBeenCalledWith(expected);
    });

  });
});
