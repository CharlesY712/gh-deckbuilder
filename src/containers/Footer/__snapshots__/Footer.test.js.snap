// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer component Container Tests should match snapshot when there are decks 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Footer
    addDecks={[Function]}
    addSelectedDecks={[Function]}
    currentDecks={
        Array [
            Object {
              "cards": Array [
                1,
                2,
                3,
                4,
                5,
                6,
                7,
                8,
              ],
              "class": "Brute",
              "deckId": 1,
              "deckName": "Hello",
            },
            Object {
              "cards": Array [
                10,
                2,
                3,
                12,
                5,
                6,
                7,
                8,
              ],
              "class": "Cragheart",
              "deckId": 12,
              "deckName": "Friend",
            },
          ]
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="saved-decks-tab"
>
          <h1
                    className="saved-decks-title"
          >
                    SAVED DECKS
          </h1>
</div>,
        <div
          className="saved-decks-container"
>
          <div>
                    <p
                              className="sign-in-request"
                    >
                              Please sign-in:
                    </p>
                    <FirebaseAuth
                              firebaseAuth={
                                        Object {
                                                  "apiKey": "AIzaSyBHV499x0Il6LN4XYA1JCExxTH_kgkg3pg",
                                                  "appName": "[DEFAULT]",
                                                  "authDomain": "gh-deckbuilder.firebaseapp.com",
                                                  "currentUser": null,
                                                }
                              }
                              uiCallback={[Function]}
                              uiConfig={
                                        Object {
                                                  "callbacks": Object {
                                                    "signInSuccessWithAuthResult": [Function],
                                                  },
                                                  "signInFlow": "popup",
                                                  "signInOptions": Array [
                                                    "google.com",
                                                    "facebook.com",
                                                  ],
                                                }
                              }
                    />
          </div>
</div>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <h1
            className="saved-decks-title"
>
            SAVED DECKS
</h1>,
          "className": "saved-decks-tab",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "SAVED DECKS",
            "className": "saved-decks-title",
          },
          "ref": null,
          "rendered": "SAVED DECKS",
          "type": "h1",
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <div>
            <p
                        className="sign-in-request"
            >
                        Please sign-in:
            </p>
            <FirebaseAuth
                        firebaseAuth={
                                    Object {
                                                "apiKey": "AIzaSyBHV499x0Il6LN4XYA1JCExxTH_kgkg3pg",
                                                "appName": "[DEFAULT]",
                                                "authDomain": "gh-deckbuilder.firebaseapp.com",
                                                "currentUser": null,
                                              }
                        }
                        uiCallback={[Function]}
                        uiConfig={
                                    Object {
                                                "callbacks": Object {
                                                  "signInSuccessWithAuthResult": [Function],
                                                },
                                                "signInFlow": "popup",
                                                "signInOptions": Array [
                                                  "google.com",
                                                  "facebook.com",
                                                ],
                                              }
                        }
            />
</div>,
          "className": "saved-decks-container",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <p
                className="sign-in-request"
>
                Please sign-in:
</p>,
              <FirebaseAuth
                firebaseAuth={
                                Object {
                                                "apiKey": "AIzaSyBHV499x0Il6LN4XYA1JCExxTH_kgkg3pg",
                                                "appName": "[DEFAULT]",
                                                "authDomain": "gh-deckbuilder.firebaseapp.com",
                                                "currentUser": null,
                                              }
                }
                uiCallback={[Function]}
                uiConfig={
                                Object {
                                                "callbacks": Object {
                                                  "signInSuccessWithAuthResult": [Function],
                                                },
                                                "signInFlow": "popup",
                                                "signInOptions": Array [
                                                  "google.com",
                                                  "facebook.com",
                                                ],
                                              }
                }
/>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Please sign-in:",
                "className": "sign-in-request",
              },
              "ref": null,
              "rendered": "Please sign-in:",
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "firebaseAuth": Object {
                  "apiKey": "AIzaSyBHV499x0Il6LN4XYA1JCExxTH_kgkg3pg",
                  "appName": "[DEFAULT]",
                  "authDomain": "gh-deckbuilder.firebaseapp.com",
                  "currentUser": null,
                },
                "uiCallback": [Function],
                "uiConfig": Object {
                  "callbacks": Object {
                    "signInSuccessWithAuthResult": [Function],
                  },
                  "signInFlow": "popup",
                  "signInOptions": Array [
                    "google.com",
                    "facebook.com",
                  ],
                },
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "div",
        },
        "type": "div",
      },
    ],
    "type": "footer",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="saved-decks-tab"
>
            <h1
                        className="saved-decks-title"
            >
                        SAVED DECKS
            </h1>
</div>,
          <div
            className="saved-decks-container"
>
            <div>
                        <p
                                    className="sign-in-request"
                        >
                                    Please sign-in:
                        </p>
                        <FirebaseAuth
                                    firebaseAuth={
                                                Object {
                                                            "apiKey": "AIzaSyBHV499x0Il6LN4XYA1JCExxTH_kgkg3pg",
                                                            "appName": "[DEFAULT]",
                                                            "authDomain": "gh-deckbuilder.firebaseapp.com",
                                                            "currentUser": null,
                                                          }
                                    }
                                    uiCallback={[Function]}
                                    uiConfig={
                                                Object {
                                                            "callbacks": Object {
                                                              "signInSuccessWithAuthResult": [Function],
                                                            },
                                                            "signInFlow": "popup",
                                                            "signInOptions": Array [
                                                              "google.com",
                                                              "facebook.com",
                                                            ],
                                                          }
                                    }
                        />
            </div>
</div>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <h1
              className="saved-decks-title"
>
              SAVED DECKS
</h1>,
            "className": "saved-decks-tab",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "SAVED DECKS",
              "className": "saved-decks-title",
            },
            "ref": null,
            "rendered": "SAVED DECKS",
            "type": "h1",
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <div>
              <p
                            className="sign-in-request"
              >
                            Please sign-in:
              </p>
              <FirebaseAuth
                            firebaseAuth={
                                          Object {
                                                        "apiKey": "AIzaSyBHV499x0Il6LN4XYA1JCExxTH_kgkg3pg",
                                                        "appName": "[DEFAULT]",
                                                        "authDomain": "gh-deckbuilder.firebaseapp.com",
                                                        "currentUser": null,
                                                      }
                            }
                            uiCallback={[Function]}
                            uiConfig={
                                          Object {
                                                        "callbacks": Object {
                                                          "signInSuccessWithAuthResult": [Function],
                                                        },
                                                        "signInFlow": "popup",
                                                        "signInOptions": Array [
                                                          "google.com",
                                                          "facebook.com",
                                                        ],
                                                      }
                            }
              />
</div>,
            "className": "saved-decks-container",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <p
                  className="sign-in-request"
>
                  Please sign-in:
</p>,
                <FirebaseAuth
                  firebaseAuth={
                                    Object {
                                                      "apiKey": "AIzaSyBHV499x0Il6LN4XYA1JCExxTH_kgkg3pg",
                                                      "appName": "[DEFAULT]",
                                                      "authDomain": "gh-deckbuilder.firebaseapp.com",
                                                      "currentUser": null,
                                                    }
                  }
                  uiCallback={[Function]}
                  uiConfig={
                                    Object {
                                                      "callbacks": Object {
                                                        "signInSuccessWithAuthResult": [Function],
                                                      },
                                                      "signInFlow": "popup",
                                                      "signInOptions": Array [
                                                        "google.com",
                                                        "facebook.com",
                                                      ],
                                                    }
                  }
/>,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Please sign-in:",
                  "className": "sign-in-request",
                },
                "ref": null,
                "rendered": "Please sign-in:",
                "type": "p",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "firebaseAuth": Object {
                    "apiKey": "AIzaSyBHV499x0Il6LN4XYA1JCExxTH_kgkg3pg",
                    "appName": "[DEFAULT]",
                    "authDomain": "gh-deckbuilder.firebaseapp.com",
                    "currentUser": null,
                  },
                  "uiCallback": [Function],
                  "uiConfig": Object {
                    "callbacks": Object {
                      "signInSuccessWithAuthResult": [Function],
                    },
                    "signInFlow": "popup",
                    "signInOptions": Array [
                      "google.com",
                      "facebook.com",
                    ],
                  },
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
            ],
            "type": "div",
          },
          "type": "div",
        },
      ],
      "type": "footer",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`Footer component Container Tests should match the snapshot when there are no decks 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Footer
    addDecks={[Function]}
    addSelectedDecks={[Function]}
    currentDecks={Array []}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="saved-decks-tab"
>
          <h1
                    className="saved-decks-title"
          >
                    SAVED DECKS
          </h1>
</div>,
        <div
          className="saved-decks-container"
>
          <div>
                    <p
                              className="sign-in-request"
                    >
                              Please sign-in:
                    </p>
                    <FirebaseAuth
                              firebaseAuth={
                                        Object {
                                                  "apiKey": "AIzaSyBHV499x0Il6LN4XYA1JCExxTH_kgkg3pg",
                                                  "appName": "[DEFAULT]",
                                                  "authDomain": "gh-deckbuilder.firebaseapp.com",
                                                  "currentUser": null,
                                                }
                              }
                              uiCallback={[Function]}
                              uiConfig={
                                        Object {
                                                  "callbacks": Object {
                                                    "signInSuccessWithAuthResult": [Function],
                                                  },
                                                  "signInFlow": "popup",
                                                  "signInOptions": Array [
                                                    "google.com",
                                                    "facebook.com",
                                                  ],
                                                }
                              }
                    />
          </div>
</div>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <h1
            className="saved-decks-title"
>
            SAVED DECKS
</h1>,
          "className": "saved-decks-tab",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "SAVED DECKS",
            "className": "saved-decks-title",
          },
          "ref": null,
          "rendered": "SAVED DECKS",
          "type": "h1",
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <div>
            <p
                        className="sign-in-request"
            >
                        Please sign-in:
            </p>
            <FirebaseAuth
                        firebaseAuth={
                                    Object {
                                                "apiKey": "AIzaSyBHV499x0Il6LN4XYA1JCExxTH_kgkg3pg",
                                                "appName": "[DEFAULT]",
                                                "authDomain": "gh-deckbuilder.firebaseapp.com",
                                                "currentUser": null,
                                              }
                        }
                        uiCallback={[Function]}
                        uiConfig={
                                    Object {
                                                "callbacks": Object {
                                                  "signInSuccessWithAuthResult": [Function],
                                                },
                                                "signInFlow": "popup",
                                                "signInOptions": Array [
                                                  "google.com",
                                                  "facebook.com",
                                                ],
                                              }
                        }
            />
</div>,
          "className": "saved-decks-container",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <p
                className="sign-in-request"
>
                Please sign-in:
</p>,
              <FirebaseAuth
                firebaseAuth={
                                Object {
                                                "apiKey": "AIzaSyBHV499x0Il6LN4XYA1JCExxTH_kgkg3pg",
                                                "appName": "[DEFAULT]",
                                                "authDomain": "gh-deckbuilder.firebaseapp.com",
                                                "currentUser": null,
                                              }
                }
                uiCallback={[Function]}
                uiConfig={
                                Object {
                                                "callbacks": Object {
                                                  "signInSuccessWithAuthResult": [Function],
                                                },
                                                "signInFlow": "popup",
                                                "signInOptions": Array [
                                                  "google.com",
                                                  "facebook.com",
                                                ],
                                              }
                }
/>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Please sign-in:",
                "className": "sign-in-request",
              },
              "ref": null,
              "rendered": "Please sign-in:",
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "firebaseAuth": Object {
                  "apiKey": "AIzaSyBHV499x0Il6LN4XYA1JCExxTH_kgkg3pg",
                  "appName": "[DEFAULT]",
                  "authDomain": "gh-deckbuilder.firebaseapp.com",
                  "currentUser": null,
                },
                "uiCallback": [Function],
                "uiConfig": Object {
                  "callbacks": Object {
                    "signInSuccessWithAuthResult": [Function],
                  },
                  "signInFlow": "popup",
                  "signInOptions": Array [
                    "google.com",
                    "facebook.com",
                  ],
                },
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "div",
        },
        "type": "div",
      },
    ],
    "type": "footer",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="saved-decks-tab"
>
            <h1
                        className="saved-decks-title"
            >
                        SAVED DECKS
            </h1>
</div>,
          <div
            className="saved-decks-container"
>
            <div>
                        <p
                                    className="sign-in-request"
                        >
                                    Please sign-in:
                        </p>
                        <FirebaseAuth
                                    firebaseAuth={
                                                Object {
                                                            "apiKey": "AIzaSyBHV499x0Il6LN4XYA1JCExxTH_kgkg3pg",
                                                            "appName": "[DEFAULT]",
                                                            "authDomain": "gh-deckbuilder.firebaseapp.com",
                                                            "currentUser": null,
                                                          }
                                    }
                                    uiCallback={[Function]}
                                    uiConfig={
                                                Object {
                                                            "callbacks": Object {
                                                              "signInSuccessWithAuthResult": [Function],
                                                            },
                                                            "signInFlow": "popup",
                                                            "signInOptions": Array [
                                                              "google.com",
                                                              "facebook.com",
                                                            ],
                                                          }
                                    }
                        />
            </div>
</div>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <h1
              className="saved-decks-title"
>
              SAVED DECKS
</h1>,
            "className": "saved-decks-tab",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "SAVED DECKS",
              "className": "saved-decks-title",
            },
            "ref": null,
            "rendered": "SAVED DECKS",
            "type": "h1",
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <div>
              <p
                            className="sign-in-request"
              >
                            Please sign-in:
              </p>
              <FirebaseAuth
                            firebaseAuth={
                                          Object {
                                                        "apiKey": "AIzaSyBHV499x0Il6LN4XYA1JCExxTH_kgkg3pg",
                                                        "appName": "[DEFAULT]",
                                                        "authDomain": "gh-deckbuilder.firebaseapp.com",
                                                        "currentUser": null,
                                                      }
                            }
                            uiCallback={[Function]}
                            uiConfig={
                                          Object {
                                                        "callbacks": Object {
                                                          "signInSuccessWithAuthResult": [Function],
                                                        },
                                                        "signInFlow": "popup",
                                                        "signInOptions": Array [
                                                          "google.com",
                                                          "facebook.com",
                                                        ],
                                                      }
                            }
              />
</div>,
            "className": "saved-decks-container",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <p
                  className="sign-in-request"
>
                  Please sign-in:
</p>,
                <FirebaseAuth
                  firebaseAuth={
                                    Object {
                                                      "apiKey": "AIzaSyBHV499x0Il6LN4XYA1JCExxTH_kgkg3pg",
                                                      "appName": "[DEFAULT]",
                                                      "authDomain": "gh-deckbuilder.firebaseapp.com",
                                                      "currentUser": null,
                                                    }
                  }
                  uiCallback={[Function]}
                  uiConfig={
                                    Object {
                                                      "callbacks": Object {
                                                        "signInSuccessWithAuthResult": [Function],
                                                      },
                                                      "signInFlow": "popup",
                                                      "signInOptions": Array [
                                                        "google.com",
                                                        "facebook.com",
                                                      ],
                                                    }
                  }
/>,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Please sign-in:",
                  "className": "sign-in-request",
                },
                "ref": null,
                "rendered": "Please sign-in:",
                "type": "p",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "firebaseAuth": Object {
                    "apiKey": "AIzaSyBHV499x0Il6LN4XYA1JCExxTH_kgkg3pg",
                    "appName": "[DEFAULT]",
                    "authDomain": "gh-deckbuilder.firebaseapp.com",
                    "currentUser": null,
                  },
                  "uiCallback": [Function],
                  "uiConfig": Object {
                    "callbacks": Object {
                      "signInSuccessWithAuthResult": [Function],
                    },
                    "signInFlow": "popup",
                    "signInOptions": Array [
                      "google.com",
                      "facebook.com",
                    ],
                  },
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
            ],
            "type": "div",
          },
          "type": "div",
        },
      ],
      "type": "footer",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
